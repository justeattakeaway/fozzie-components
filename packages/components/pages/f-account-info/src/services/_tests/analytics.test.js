import AccountInfoAnalyticsService from '../analytics';

let gtmMock;
let analytics;

const pushEventSpy = jest.fn();

describe('Analytics', () => {
    beforeEach(() => {
        gtmMock = {
            pushEvent: pushEventSpy
        };

        analytics = new AccountInfoAnalyticsService(gtmMock);
    });

    afterEach(() => {
        jest.clearAllMocks();
    });

    describe('trackFormSubmission ::', () => {
        describe('address has not been updated', () => {
            it('should push submit event only ', () => {
                // Act
                analytics.trackFormSubmission(false);

                // Assert
                expect(pushEventSpy).toMatchSnapshot();
            });
        });

        describe('address has been updated', () => {
            it('should push submit and address changed events', () => {
                // Act
                analytics.trackFormSubmission(true);

                // Assert
                expect(pushEventSpy).toMatchSnapshot();
            });
        });
    });
});
