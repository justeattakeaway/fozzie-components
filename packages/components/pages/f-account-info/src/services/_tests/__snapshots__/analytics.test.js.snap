// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Analytics trackFormSubmission :: address has been updated should push submit and address changed events 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "action": "save_accountinfo_changes",
        "category": "account",
        "event": "trackEvent",
        "label": "submit",
      },
    ],
    Array [
      Object {
        "action": "account info",
        "category": "my account",
        "event": "trackEvent",
        "label": "address change intent",
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`Analytics trackFormSubmission :: address has not been updated should push submit event only  1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "action": "save_accountinfo_changes",
        "category": "account",
        "event": "trackEvent",
        "label": "submit",
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
